crates/action-pipeline/src/subscribers/webhooks_subscriber.rs

```diff
@@ -1,29 +1,44 @@
+use std::sync::Arc;
+
 use crate::event_emitter::{Event, Subscriber};
 use async_trait::async_trait;
 use moon_notifier::WebhooksNotifier;
-use tracing::debug;
+use moon_process::ProcessRegistry;
+use tracing::{debug, error};
 
 pub struct WebhooksSubscriber {
     notifier: WebhooksNotifier,
+    acknowledge: bool,
+    receiver: Arc<ProcessRegistry>,
 }
 
 impl WebhooksSubscriber {
-    pub fn new(url: &str) -> Self {
+    pub fn new(url: &str, acknowledge: &bool, receiver: Arc<ProcessRegistry>) -> Self {
         WebhooksSubscriber {
             notifier: WebhooksNotifier::new(url.to_owned()),
+            acknowledge: acknowledge.to_owned(),
+            receiver,
         }
     }
 }
 
 #[async_trait]
 impl Subscriber for WebhooksSubscriber {
     async fn on_emit<'data>(&mut self, event: &Event<'data>) -> miette::Result<()> {
-        self.notifier.notify(event.get_type(), event).await?;
+        if self.acknowledge {
+            let _ = self.notifier.wait_for_requests().await;
+            if !self.notifier.verified {
+                self.receiver.terminate_running();
```

Comment:
I think if you update `notify()` to return an error for non-200, it would just bubble up and shutdown the pipeline automatically.